libs/x-content/src/main/java/org/elasticsearch/xcontent/internal/XContentProviderLocator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.xcontent.internal;
+
+import org.elasticsearch.core.internal.provider.ProviderLocator;
+import org.elasticsearch.xcontent.spi.XContentProvider;
+
+import java.util.ServiceConfigurationError;
+
+/**
+ * A provider locator for finding the {@link XContentProvider}.
+ */
+public final class XContentProviderLocator {
```

Comment:
Do we need a separate class, or could this simply be inside XContentProvider? It's odd that we have "INSTANCE" which is not this class itself, but XContentProvider, and there we *also* have instance which just pulls from this class.

Reply:
I folder this into an inner class in the XContentProvider interface ( since we just want a single static instance ).